src/clincoded/static/components/provisional_classification/snapshots.js

```diff
@@ -107,24 +107,38 @@ class CurationSnapshots extends Component {
 
     /**
      * Method to render the button that allows users to publish/unpublish an approved classification
-     * The target="_self" attribute is included to ensure page is reloaded (to restart lifecycle of ProvisionalClassification component)
      * @param {object} resourceParent - The parent object of the classification in a snapshot
      * @param {string} snapshotUUID - The UUID of the source snapshot
      * @param {boolean} publishClassification - The published status of the classification (per the source snapshot)
      */
     renderPublishLink(resourceParent, snapshotUUID, publishClassification) {
-        if (resourceParent && resourceParent.uuid) {
-            if (!publishClassification) {
+        let classData = 'btn btn-default publish-link-item';
+        let eventType = 'publish';
+        let buttonText = 'Publish Summary';
+
+        if (publishClassification) {
+            classData += ' unpublish';
+            eventType = 'unpublish';
+            buttonText = 'Unpublish Summary';
+        }
+
+        // If already within the approval process, present publish link as a button (that triggers a state update in a parent component)
+        if (typeof window !== "undefined" && window.location && window.location.pathname && window.location.pathname.indexOf('provisional-classification') > -1) {
```

Comment:
You can instead pass parent's `context` prop to this child component and use it like:
```
...
const context = this.props.context;
...
if (context && context.name === 'provisional-classification')
...
```

Reply:
Yea, passing a prop indicating the parent component would probably be a better solution.

Reply:
I'll plan to take care of this in another ticket.

Reply:
I am addressing this in my branch upon merging yours.